src/main/java/zeronote/userinterface/CliUserInterface.java

```diff
@@ -31,6 +31,60 @@ private void saveState() throws ZeroNoteException {
         storage.saveToFile(appState);
     }
 
+    private void printPrompt() {
+        String modeString = "";
+        String notebookTitle;
+        String sectionTitle;
+        String pageTitle;
+        try {
+            notebookTitle = appState.getCurrentNotebook().getTitle();
+        } catch (NullPointerException e) {
+            notebookTitle = "";
+        }
+        try {
+            sectionTitle = appState.getCurrentSection().getTitle();
+        } catch (NullPointerException e) {
+            sectionTitle = "";
+        }
+        try {
+            pageTitle = appState.getCurrentPage().getTitle();
+        } catch (NullPointerException e) {
+            pageTitle = "";
+        }
+        if (notebookTitle.length() > 10) {
+            notebookTitle = notebookTitle.substring(0, 7) + "...";
+        }
+        if (sectionTitle.length() > 10) {
+            sectionTitle = sectionTitle.substring(0, 7) + "...";
+        }
+        if (pageTitle.length() > 10) {
+            pageTitle = pageTitle.substring(0, 7) + "...";
+        }
```

Comment:
since we're checking for whether the length is greater than 10, why do we make a substring that ends at 7 o.o

Reply:
Because we add `...` which makes it 10. If we check for length greater than, say, 7, and then take the substring from 0 to 7, then if the title were something like `Buoyancy`, it would be turned into `Buoyanc...` which is actually more characters. I've seen this UI glitch before in actual, production software...